src/uu/tr/src/operation.rs

```diff
@@ -708,9 +708,16 @@ where
         let filtered = buf.iter().filter_map(|&c| translator.translate(c));
         output_buf.extend(filtered);
 
-        output
-            .write_all(&output_buf)
-            .map_err_context(|| get_message("tr-error-write-error"))?;
+        if let Err(err) = output.write_all(&output_buf) {
+            // Treat broken pipe as a successful termination, which is the
```

Comment:
> successful termination

Not really:

```console
$ yes | tr y n | head
n
n
n
n
n
n
n
n
n
n
$ echo ${PIPESTATUS[@]}
141 141 0
```

141 is an artifact of bash.  The expected behaviour is to die by `SIGPIPE` which is distinguishable from `exit(141)`

Reply:
Indeed, thanks for the correction. The solution should really be restoring the default action of the SIGPIPE signal (Rust ignores it for [reasons](https://github.com/rust-lang/rust/issues/62569)).

Reply:
https://github.com/uutils/coreutils/pull/8312. cc @cakebaker